examples/composer-shared-vpc/composer_v1_pvt_shared_vpc/main.tf

```diff
@@ -0,0 +1,75 @@
+
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  gke_agent_sa      = "service-${data.google_project.project.number}@container-engine-robot.iam.gserviceaccount.com"
+  cloud_services_sa = "${data.google_project.project.number}@cloudservices.gserviceaccount.com"
+  composer_agent_sa = "service-${data.google_project.project.number}@cloudcomposer-accounts.iam.gserviceaccount.com"
+  composer_worker_sa_email = (
+    var.composer_service_account_create
+    ? (
+      length(google_service_account.composer_worker_sa) > 0
+      ? google_service_account.composer_worker_sa[0].email
+      : null
+    )
+    : var.composer_service_account
+  )
+  subnet_primary_cidr_range    = data.google_compute_subnetwork.subnetwork.ip_cidr_range
+  subnet_pod_cidr_range        = [for secondary_range in data.google_compute_subnetwork.subnetwork.secondary_ip_range : secondary_range.ip_cidr_range if secondary_range.range_name == var.pod_ip_allocation_range_name][0]
+  gke_agents_permissions       = var.assign_robot_sa_permissions ? ["roles/compute.networkUser", "roles/compute.securityAdmin", "roles/container.hostServiceAgentUser"] : []
+  composer_agents_permissions  = var.assign_robot_sa_permissions ? ["roles/compute.networkUser", "roles/composer.sharedVpcAgent"] : []
+  cloudservices_sa_permissions = var.assign_robot_sa_permissions ? ["roles/compute.networkUser"] : []
+}
+
+data "google_compute_subnetwork" "subnetwork" {
+  name    = var.subnetwork
+  project = var.network_project_id
+  region  = var.subnetwork_region
+}
+
+
+module "composer-v1-private" {
```

Comment:
I would add support for `software_config` (useful to specify env variables) and variables to setup private configuration (example: cloud_sql_ipv4_cidr)

Reply:
Done, added software config.